test/tests.py

```diff
@@ -0,0 +1,142 @@
+from operations import create_index, check_index, create_document, delete_document, get_document, delete_index
+from operations import check_document
+from http import HTTPStatus
+import unittest
+import os
+import logging
+import time
+import requests
+
+logger = logging.getLogger(__name__)
+
+# Tests will say which line the test failed at and what was the result of the execution, but better logging can be done.
+
+
+def retry_request(request, args=(), max_attempts=10, delay=0.5, expectedStatusCode=None):
+    for attempt in range(1, max_attempts + 1):
+        try:
+            result = request(*args)
+            if result.status_code == expectedStatusCode:
+                return result
+            else:
+                logger.error(f"Status code returned: {result.status_code} did not"
+                             f" match the expected status code: {expectedStatusCode}")
+        except Exception:
+            logger.error(f"Trying again in {delay} seconds")
```

Comment:
I would break out of this for loop in this case. This is an unknown exception and a retry will probably fail the same way. The reason you have this retry in is because of eventual consistency. Other exceptions should not be handled as a retry. Simply log error (as you are doing), then raise. Remove the sleep.